receiver/statsdreceiver/receiver.go

```diff
@@ -110,10 +122,13 @@ func (r *statsdReceiver) Start(ctx context.Context, _ component.Host) error {
 				batchMetrics := r.parser.GetMetrics()
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
-
-					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+					numPoints := batch.Metrics.DataPointCount()
+					r.obsrecv.StartMetricsOp(ctx)
```

Comment:
This function should take the `batchCtx` as the input context and the returned context should be used in `Flush`, and `EndMetricsOp`

Reply:
fixed PTAL